storage/uri_test.go

```diff
@@ -17,31 +17,107 @@ import (
 )
 
 type otherURI struct {
+	path string
 	fyne.URI
```

Comment:
With the URI implemented using just path I think this embedded `URI` may just end up hiding potential issues, for example if one of the methods was not implemented?